Notation.Plugin.AzureKeyVault/Command/GenerateSignature.cs

```diff
@@ -0,0 +1,59 @@
+using System.Text.Json;
+using Notation.Plugin.AzureKeyVault.Certificate;
+using Notation.Plugin.AzureKeyVault.Client;
+using Notation.Plugin.Protocol;
+
+namespace Notation.Plugin.AzureKeyVault.Command
+{
+    /// <summary>
+    /// Implementation of describe-key command.
+    /// </summary>
+    public class GenerateSignature : IPluginCommand
+    {
+        public async Task<object> RunAsync(string inputJson)
+        {
+            // Parse the input
+            GenerateSignatureRequest? input = JsonSerializer.Deserialize<GenerateSignatureRequest>(inputJson);
+            if (input == null)
+            {
+                throw new ValidationException("Invalid input");
+            }
+
+            var akvClient = new KeyVaultClient(input.KeyId);
+
+            // Extract signature algorithm from the certificate
+            var leafCert = await akvClient.GetCertificate();
+            var keySpec = KeySpecUtils.ExtractKeySpec(leafCert);
+            var signatureAlgorithm = SignatureAlgorithmHelper.FromKeySpec(keySpec);
+
+            // Sign
+            var signature = await akvClient.Sign(signatureAlgorithm, input.Payload);
```

Comment:
I might miss it. But, should we hash the `input.Payload` before signing?

Reply:
The hash was done in `_cryptoClient.Value.SignDataAsync(algorithm, payload)`, so we don't need to do it.

Reply:
Ah, nice.